src/OptionT.ts

```diff
@@ -466,30 +484,201 @@ export interface OptT<T> {
    * @param {(val: T) => boolean} condition
    * @returns {boolean}
    */
-  contains(condition: (val: T) => boolean): boolean;
+  abstract contains(condition: (val: T) => boolean): boolean;
 }
 
+
 /**
- * Wrapper for `some` and `none` functions.
+ * A class representing the `None`-type variant of the `OptionT` type.
+ *
+ * Instances of this class contain no internal value.  They simply wrap the concept of 'nothing'
+ * inside the same `OptionT` API defined by [[OptT]].
  */
-export namespace OptionT {
-  /**
-   * Returns a `None` value if `val` is `null` or `undefined`, otherwise returns a `Some`
-   * value containing `val`.
-   *
-   * @param {T} val
-   * @returns {OptT<any>}
-   */
-  export function some<T>(val: T) {
-    return getSome(val);
+class None<T> extends OptionT<T> {
+  constructor() {
+    super();
   }
 
-  /**
-   * Returns a `None` value.
-   *
-   * @returns {OptT<any>}
-   */
-  export function none() {
-    return getNone();
+  isSome(): boolean {
+    return false;
+  }
+
+  isNone(): boolean {
+    return true;
+  }
+
+  toString(): string {
+    return 'None()';
+  }
+
+  expect(message: string): never {
+    throw new Error(message);
+  }
+
+  unwrap(): never {
+    throw new Error('Called unwrap on a None value.');
+  }
+
+  unwrapOr(other: T): T {
+    return other;
+  }
+
+  unwrapOrElse(func: () => T): T {
+    return func();
+  }
+
+  map<U>(func: (val: T) => U): OptionT<U> {
+    return <OptionT<U>>new None();
+  }
+
+  mapOr<U>(other: U, func: (val: T) => U): U {
+    return other;
+  }
+
+  mapOrElse<U>(other: () => U, func: (val: T) => U): U {
+    return other();
+  }
+
+  and<U>(other: OptionT<U>): OptionT<U> {
+    return <OptionT<U>>new None();
+  }
+
+  flatMap<U>(func: (val: T) => OptionT<U>): OptionT<U> {
+    return <OptionT<U>>new None();
+  }
+
+  or(other: OptionT<T>): OptionT<T> {
+    return <OptionT<T>>other;
```

Comment:
nitpick: is this cast necessary?  Shouldn't we just be able to `return other;`?

Reply:
I will look, but this hasnt changed really from master - except its OptionT instead of OptT.
https://github.com/iwburns/nullshield/blob/master/src/option/none.ts#L58

Reply:
yeah, I gotcha.  If you want to leave it in that's cool, but I think it didn't really need to be in there to begin with.